bindata/etcd/cluster-restore.sh

```diff
@@ -140,3 +140,5 @@ else
   mv "${MANIFEST_STOPPED_DIR}/etcd-pod.yaml" "${MANIFEST_DIR}/etcd-pod.yaml"
 fi
 
+# This ensures kubelet does not get stuck on reporting status of the static pod, see OCPBUGS-42133
+systemctl restart kubelet
```

Comment:
If this can happen when moving etcd yaml in cluster-restore.sh; is can also happen in installer pod(as in installer pod, it also trigger etcd update by moving yaml from static-resource-manifests to manifests dir)

Reply:
it can, yes

Reply:
could u show me the issue or jira bug, I have once encountered in the static problem in single node, but not HA mode

Reply:
we have something similar tracked under https://issues.redhat.com/browse/OCPBUGS-36867

Reply:
- After reading the issue, I found the kubelet has reported some logs that are very familiar 
```
./ip-10-0-110-52.us-east-2.compute.internal_logs_kubelet:Jul 16 06:52:29.541102 ip-10-0-110-52 kubenswrapper[2242]: E0716 06:52:29.541065    2242 mirror_client.go:138] "Failed deleting a mirror pod" err="Timeout: request did not complete within requested timeout - context deadline exceeded" pod="openshift-kube-controller-manager/kube-controller-manager-ip-10-0-110-52.us-east-2.compute.internal"

./ip-10-0-110-52.us-east-2.compute.internal_logs_kubelet:Jul 16 06:53:03.649070 ip-10-0-110-52 kubenswrapper[2242]: E0716 06:53:03.649033    2242 mirror_client.go:138] "Failed deleting a mirror pod" err="Timeout: request did not complete within requested timeout - context deadline exceeded" pod="openshift-kube-controller-manager/kube-controller-manager-ip-10-0-110-52.us-east-2.compute.internal"

./ip-10-0-110-52.us-east-2.compute.internal_logs_kubelet:Jul 16 06:53:37.759141 ip-10-0-110-52 kubenswrapper[2242]: E0716 06:53:37.759100    2242 mirror_client.go:138] "Failed deleting a mirror pod" err="Timeout: request did not complete within requested timeout - context deadline exceeded" pod="openshift-kube-controller-manager/kube-controller-manager-ip-10-0-110-52.us-east-2.compute.internal"

./ip-10-0-110-52.us-east-2.compute.internal_logs_kubelet:Jul 16 06:54:12.911545 ip-10-0-110-52 kubenswrapper[2242]: E0716 06:54:12.911510    2242 mirror_client.go:138] "Failed deleting a mirror pod" err="Timeout: request did not complete within requested timeout - context deadline exceeded" pod="openshift-kube-controller-manager/kube-controller-manager-ip-10-0-110-52.us-east-2.compute.internal"

```
- these logs indicate apiserver is not available that time, and when apiserver is down，and the haproxy returns EOF，this will trigger the kubelet client  sleep and trottle, which will make the kubelet pull static pods very slow as it will stuck in the volumemanager, I have met this problem before, https://github.com/kubernetes/kubernetes/issues/126711

Hope this gives you some hints

Reply:
yep we know about this, but on SNO there's also little you can do when either API or etcd is down. 
cc @rphillips

Reply:
> yep we know about this, but on SNO there's also little you can do when either API or etcd is down. cc @rphillips

maybe in SNO, we should not depend on the haproxy, but let the kubelet connect apiserver on 127.0.0.1

Reply:
maybe @eggfoobar can help you on this specifically, I'm not a routing expert for the apiserver on SNO